libraries/ESP8266mDNS/src/LEAmDNS2Host.cpp

```diff
@@ -0,0 +1,1364 @@
+                                     bool p_bAnnounce /*= true*/)
+{
+
+    return _announceService(*p_pService, p_bAnnounce);
+}
+
+
+/*
+    clsLEAmDNS2_Host::restart
+*/
+bool clsLEAMDNSHost::restart(void)
+{
+    return (_resetProbeStatus(true));   // Stop and restart probing
+}
+
+
+/*
+    clsLEAMDNSHost_Legacy::enableArduino
+*/
+clsLEAMDNSHost::clsService* clsLEAMDNSHost::enableArduino(uint16_t p_u16Port,
+        bool p_bAuthUpload /*= false*/)
+{
+    clsLEAMDNSHost::clsService* svc = addService("arduino", "arduino", "tcp", p_u16Port);
+    if (svc)
+    {
+        if ((!svc->addServiceTxt("tcp_check", "no"))
+                || (!svc->addServiceTxt("ssh_upload", "no"))
+                || (!svc->addServiceTxt("board", STRINGIZE_VALUE_OF(ARDUINO_BOARD)))
+                || (!svc->addServiceTxt("auth_upload", (p_bAuthUpload) ? "yes" : "no")))
+        {
+            removeService(svc);
+            svc = 0;
+        }
+    }
+    return svc;
+}
+
+
+/*
+    P R O T E C T E D
+*/
+
+/*
+
+    BACKBONE
+
+*/
+
+/*
+    clsLEAmDNS2_Host::_allocBackbone
+
+*/
+UdpContext* clsLEAMDNSHost::_allocBackbone(void)
+{
+    UdpContext* pUDPContext = 0;
+
+    if (!clsBackbone::sm_pBackbone)
+    {
+        // Not yet created
+        clsBackbone::sm_pBackbone = new clsBackbone;
+        DEBUG_EX_INFO(DEBUG_OUTPUT.printf_P(PSTR("%s _allocBackbone: Created backbone.\n"), _DH()););
+
+        if ((clsBackbone::sm_pBackbone) &&
+                (!clsBackbone::sm_pBackbone->init()))
+        {
+            DEBUG_EX_ERR(DEBUG_OUTPUT.printf_P(PSTR("%s _allocBackbone: FAILED to init backbone!\n"), _DH()););
+
+            delete clsBackbone::sm_pBackbone;
+            clsBackbone::sm_pBackbone = 0;
+        }
+    }
+    if (clsBackbone::sm_pBackbone)
+    {
+        pUDPContext = clsBackbone::sm_pBackbone->addHost(this);
+    }
+    DEBUG_EX_INFO(DEBUG_OUTPUT.printf_P(PSTR("%s _allocBackbone: %s to add host to backbone.\n"), _DH(), (pUDPContext ? "Succeeded" : "FAILED")););
+    return pUDPContext;
+}
+
+/*
+    clsLEAmDNS2_Host::_releaseBackbone
+
+*/
+bool clsLEAMDNSHost::_releaseBackbone(void)
+{
+    bool    bResult = false;
+
+    if ((clsBackbone::sm_pBackbone) &&
+            ((bResult = clsBackbone::sm_pBackbone->removeHost(this))) &&
+            (0 == clsBackbone::sm_pBackbone->hostCount()))
+    {
+        delete clsBackbone::sm_pBackbone;
+        clsBackbone::sm_pBackbone = 0;
+        DEBUG_EX_INFO(DEBUG_OUTPUT.printf_P(PSTR("%s _releaseBackbone: Released backbone."), _DH()););
+    }
+    DEBUG_EX_INFO(DEBUG_OUTPUT.printf_P(PSTR("%s _releaseBackbone: %s to remove host from backbone."), _DH(), (bResult ? "Succeeded" : "FAILED")););
+    return bResult;
+}
+
+
+/*
+
+    MULTICAST GROUPS
+
+*/
+
+/*
+    clsLEAmDNS2_Host::_joinMulticastGroups
+*/
+bool clsLEAMDNSHost::_joinMulticastGroups(void)
+{
+    bool    bResult = false;
+
+    // Join multicast group(s)
+    for (netif* pNetIf = netif_list; pNetIf; pNetIf = pNetIf->next)
```

Comment:
> However, it's considered good practice to keep the braces if the instruction is more than one line, such as this case with a large if().

I updated, but I'd be pleased to know where these good practices are defined. What is unclear in there ?
```
    for (blah)
        if (blah)
        {
        }
```
On top of that, gcc is now able to warn when tabulation is not right and leads to human interpretation error.

Reply:
Consider this trivial variant:
```
    for (blah);
        if (blah)
        {
        }
```
In fact, it is precisely because of such cases that most coding guides will say "always use braces around the body, period". There are other reasons for this, such as what happens when you comment out the 1-line body of the if, then the "body" becomes the next line. From that pov, braces are a must in all cases. This is an age-old discussion, you can google [historic cases](https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492) or even [recent answers](https://softwareengineering.stackexchange.com/questions/16528/single-statement-if-block-braces-or-no). Sorry, I don't mean to lecture.
Me, we use Allman indent, so I'm ok with leaving the braces out for single-line bodies and leaving an empty line after, it improves readability in such cases, but I don't feel strongly about it. However, for multi-line bodies, braces should always be used regardless of whether it's a single or multiple statement body.

Reply:
Like I said, gcc is now able to detect such error:

```
int main ()
{
    for (int i = 0; i < 10; i++);
        if (i)
        {
        }
```

```
x.cc: In function ‘int main()’:
x.cc:13:5: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
   13 |     for (int i = 0; i < 10; i++);
      |     ^~~
x.cc:14:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
   14 |         if (i)
      |         ^~
x.cc:14:13: error: ‘i’ was not declared in this scope
   14 |         if (i)
      |             ^
```

Even with older version of gcc and good practice, you can make mistakes:
```
   for (blah);
   {
        if (blah)
        {
        }
   }
```